ts/screens/messages/paginated/MessageRouterScreen.tsx

```diff
@@ -0,0 +1,165 @@
+import * as allPaginated from "../../../store/reducers/entities/messages/allPaginated";
+import {
+  Cursor,
+  getCursors
+} from "../../../store/reducers/entities/messages/allPaginated";
+
+type OwnProps = NavigationInjectedProps<{ messageId: UIMessageId }>;
+type Props = OwnProps &
+  ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps>;
+
+/**
+ * Choose the screen where to navigate, based on the message content.
+ * TODO: refactor to return an action for _dispatch_
+ * @param message
+ * @param messageDetails
+ */
+const navigateToScreenHandler =
+  (message: UIMessage, messageDetails: UIMessageDetails) =>
+  (dispatch: OwnProps["navigation"]["dispatch"]) => {
+    if (euCovidCertificateEnabled && messageDetails.euCovidCertificate) {
+      navigateBack();
+      navigateToEuCovidCertificateDetailScreen({
+        authCode: messageDetails.euCovidCertificate
+          .authCode as EUCovidCertificateAuthCode,
+        messageId: message.id
+      });
+    } else {
+      navigateBack();
+      dispatch(
+        navigateToPaginatedMessageDetailScreenAction({
+          message,
+          messageDetails
+        })
+      );
+    }
+  };
+
+const MessageRouterScreen = ({
+  cancel,
+  cursors,
+  loadMessageDetails,
+  loadPreviousPage,
+  reloadPage,
+  maybeMessage,
+  maybeMessageDetails,
+  messageId,
+  navigation
+}: Props): React.ReactElement => {
+  // used to automatically dispatch loadMessages if the pot is not some at the first rendering
+  // (avoid displaying error at the first frame)
+  const firstRendering = useRef(true);
+  const isLoading = !pot.isError(maybeMessageDetails);
+  const errorMessage = pot.isError(maybeMessageDetails)
+    ? maybeMessageDetails.error
+    : undefined;
+
+  const tryLoadMessageDetails = useCallback(() => {
+    if (maybeMessage === undefined) {
+      if (pot.isNone(cursors)) {
+        // nothing in the collection, refresh
+        reloadPage();
+      } else if (pot.isSome(cursors)) {
+        // something in the collection, get the new ones only
+        loadPreviousPage(cursors.value.previous);
+      }
+    }
+    loadMessageDetails(messageId);
+  }, [
+    maybeMessage,
+    cursors,
+    messageId,
+    loadMessageDetails,
+    loadPreviousPage,
+    reloadPage
+  ]);
+
+  useEffect(() => {
+    // message in the list and its details loaded: green light
+    if (isStrictSome(maybeMessageDetails) && maybeMessage !== undefined) {
+      navigateToScreenHandler(
+        maybeMessage,
+        maybeMessageDetails.value
+      )(navigation.dispatch);
+      return;
+    }
+    if (firstRendering.current) {
+      tryLoadMessageDetails();
+      // eslint-disable-next-line functional/immutable-data
+      firstRendering.current = false;
+    }
+  }, [
+    loadMessageDetails,
+    maybeMessage,
+    maybeMessageDetails,
+    messageId,
+    navigation,
+    tryLoadMessageDetails
+  ]);
+
+  return (
+    <LoadingErrorComponent
+      errorText={errorMessage}
```

Comment:
I believe we should not show any technical details to the end user, even for security reasons

<img src="https://user-images.githubusercontent.com/822471/147102359-4c00108e-ce89-49d8-8c30-a49b8a313dd4.png" height="600"/>

My suggestion is about to show a generic one
```suggestion
      errorText={I18n.t("global.genericError")}
```

<img src="https://user-images.githubusercontent.com/822471/147102724-2eadda1e-888b-457f-a7aa-c2fcc577c1a8.png" height="600"/>